Scripts/Modules/SnappingModule/SnappingModule.cs

```diff
@@ -155,20 +148,82 @@ struct SnappingDirection
 			}
 		};
 
-		bool m_DisableAll;
+		[Serializable]
+		class Preferences
+		{
+			[SerializeField]
+			bool m_DisableAll;
+
+			// Snapping Modes
+			[SerializeField]
+			bool m_GroundSnappingEnabled;
+			[SerializeField]
+			bool m_SurfaceSnappingEnabled;
+
+			// Modifiers (do not require reset on value change)
+			[SerializeField]
```

Comment:
This doesn't preserve the defaults. Without any settings saved, the following should default to enabled:

- ground snapping
- surface snapping
- manipulator snapping
- direct snapping